src/batch/src/executor/log_row_seq_scan.rs

```diff
@@ -0,0 +1,277 @@
+                table,
+                scan_ranges,
+                log_store_seq_scan_node.old_epoch.clone().unwrap(),
+                log_store_seq_scan_node.new_epoch.clone().unwrap(),
+                chunk_size as usize,
+                source.plan_node().get_identity().clone(),
+                metrics,
+            )))
+        })
+    }
+}
+impl<S: StateStore> Executor for LogRowSeqScanExecutor<S> {
+    fn schema(&self) -> &Schema {
+        &self.schema
+    }
+
+    fn identity(&self) -> &str {
+        &self.identity
+    }
+
+    fn execute(self: Box<Self>) -> BoxedDataChunkStream {
+        self.do_execute().boxed()
+    }
+}
+
+impl<S: StateStore> LogRowSeqScanExecutor<S> {
+    #[try_stream(ok = DataChunk, error = BatchError)]
+    async fn do_execute(self: Box<Self>) {
+        let Self {
+            chunk_size,
+            identity,
+            metrics,
+            table,
+            scan_ranges,
+            old_epoch,
+            new_epoch,
+            schema,
+        } = *self;
+        let table = std::sync::Arc::new(table);
+
+        // Create collector.
+        let histogram = metrics.as_ref().map(|metrics| {
+            metrics
+                .executor_metrics()
+                .row_seq_scan_next_duration
+                .with_guarded_label_values(&metrics.executor_labels(&identity))
+        });
+        // Range Scan
+        // WARN: DO NOT use `select` to execute range scans concurrently
+        //       it can consume too much memory if there're too many ranges.
+        for range in scan_ranges {
+            let stream = Self::execute_range(
+                table.clone(),
+                range,
+                old_epoch.clone(),
+                new_epoch.clone(),
+                chunk_size,
+                histogram.clone(),
+                Arc::new(schema.clone()),
+            );
+            #[for_await]
+            for chunk in stream {
+                let chunk = chunk?;
+                yield chunk;
+            }
+        }
+    }
+
+    #[try_stream(ok = DataChunk, error = BatchError)]
+    async fn execute_range(
+        table: Arc<StorageTable<S>>,
+        scan_range: ScanRange,
+        old_epoch: BatchQueryEpoch,
+        new_epoch: BatchQueryEpoch,
+        chunk_size: usize,
+        histogram: Option<impl Deref<Target = Histogram>>,
+        schema: Arc<Schema>,
+    ) {
+        let ScanRange {
+            pk_prefix,
+            next_col_bounds,
+        } = scan_range;
+
+        let order_type = table.pk_serializer().get_order_types()[pk_prefix.len()];
+        let (start_bound, end_bound) = (next_col_bounds.0, next_col_bounds.1);
+        assert!(pk_prefix.len() < table.pk_indices().len());
+        // Range Scan.
+        let iter = table
+            .batch_iter_log_with_pk_bounds(
+                old_epoch.into(),
+                new_epoch.into(),
+                &pk_prefix,
+                (
+                    match start_bound {
```

Comment:
Actually there is no need to pass the scan range as input, because we assume that the range is always `ScanRange::Full()` here. 

We can just create the bound without doing the match in start_bound and end_bound, and the `pk_prefix` is always `OwnedRow::default()`.